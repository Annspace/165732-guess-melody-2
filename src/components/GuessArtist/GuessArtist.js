import React, {PureComponent} from "react";
import PropTypes from "prop-types";

class GuessArtist extends PureComponent {
  constructor(props) {
    super(props);
  }
  render() {
    const {answers, questionText, onClickAnswer, screenIndex} = this.props;
    return (
      <section className="game game--artist">
        <header className="game__header">
          <a className="game__back" href="#">
            <span className="visually-hidden">Сыграть ещё раз</span>
            <img className="game__logo" src="img/melody-logo-ginger.png" alt="Угадай мелодию"/>
          </a>

          {/* <svg xmlns="http://www.w3.org/2000/svg" className="timer" viewBox="0 0 780 780">*/}
          {/*  <circle className="timer__line" cx="390" cy="390" r="370"*/}
          {/*    style="filter: url(#blur); transform: rotate(-90deg) scaleY(-1); transform-origin: center"/>*/}
          {/* </svg>*/}

          <div className="timer__value" xmlns="http://www.w3.org/1999/xhtml">
            <span className="timer__mins">05</span>
            <span className="timer__dots">:</span>
            <span className="timer__secs">00</span>
          </div>

          <div className="game__mistakes">
            <div className="wrong"/>
            <div className="wrong"/>
            <div className="wrong"/>
          </div>
        </header>

        <section className="game__screen">
          <h2 className="game__title">{questionText}</h2>
          <div className="game__track">
            <div className="track">
              <button className="track__button track__button--play" type="button"/>
              <div className="track__status">
                <audio/>
              </div>
            </div>
          </div>

          <form onChange={onClickAnswer} className="game__artist">
            {
              answers.map((answer, index) => {
                return (
                  <div className="artist" key={screenIndex + `-answer-` + index}>
                    <input className="artist__input visually-hidden" type="radio" name="answer" value={`answer-` + (index + 1)}
                      id={`answer-` + (index + 1)}/>
                    <label className="artist__name" htmlFor={`answer-` + (index + 1)}>
                      <img className="artist__picture" src={answer.picture} alt={answer.artist}/>
                      {answer.artist}
                    </label>
                  </div>
                );
              })
            }
          </form>
        </section>
      </section>
    );
  }
}

GuessArtist.propTypes = {
  answers: PropTypes.array.isRequired,
  questionText: PropTypes.string.isRequired,
  onClickAnswer: PropTypes.func.isRequired,
  screenIndex: PropTypes.number.isRequired,
};

export default GuessArtist;
