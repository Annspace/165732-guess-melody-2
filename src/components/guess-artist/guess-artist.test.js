import React from 'react';
import renderer from 'react-test-renderer';
import GuessArtist from "./guess-artist.jsx";
import mockData from "../../mocks/questions";

function createNodeMock(element) {
  if (element.type === `audio`) {
    // This is your fake DOM node for <p>.
    // Feel free to add any stub methods, e.g. focus() or any
    // other methods necessary to prevent crashes in your components.
    return {
      createRef() {}
    };
  }
  // You can return any object from this method for any type of DOM component.
  // React will use it as a ref instead of a DOM node when snapshot testing.
  return null;
}


it(`renders correctly`, () => {
  const options = {createNodeMock};
  const clickAnswer = jest.fn();
  const GuessArtistComponent = renderer
    .create(<GuessArtist
      screenIndex={0}
      questionText={mockData.questionsData[0].questionText}
      onClickAnswer={clickAnswer}
      answers={mockData.questionsData[0].answers}
      song={mockData.questionsData[0].song}/>, options).toJSON();
  expect(GuessArtistComponent).toMatchSnapshot();
});
