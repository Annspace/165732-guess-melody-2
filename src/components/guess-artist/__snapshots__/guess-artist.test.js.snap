// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        05
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio
            src="https://es31-server.appspot.com/guess-melody/static/music/Whaling_City.mp3"
          />
        </div>
      </div>
    </div>
    <form
      className="game__artist"
      onChange={[MockFunction]}
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-1"
          name="answer"
          type="radio"
          value="answer-1"
        />
        <label
          className="artist__name"
          htmlFor="answer-1"
        >
          <img
            alt="Quincas Moreira"
            className="artist__picture"
            src="https://es31-server.appspot.com/guess-melody/static/artist/Quincas_Moreira.jpg"
          />
          Quincas Moreira
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-2"
          name="answer"
          type="radio"
          value="answer-2"
        />
        <label
          className="artist__name"
          htmlFor="answer-2"
        >
          <img
            alt="Density & Time"
            className="artist__picture"
            src="https://es31-server.appspot.com/guess-melody/static/artist/Density_n_Time.jpg"
          />
          Density & Time
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-3"
          name="answer"
          type="radio"
          value="answer-3"
        />
        <label
          className="artist__name"
          htmlFor="answer-3"
        >
          <img
            alt="Endless Love"
            className="artist__picture"
            src="https://es31-server.appspot.com/guess-melody/static/artist/Endless_Love.jpg"
          />
          Endless Love
        </label>
      </div>
    </form>
  </section>
</section>
`;
